src/platform/Linux/DnssdImpl.h

```diff
@@ -145,12 +148,28 @@ class MdnsAvahi
         AvahiProtocol mTransport;
         AvahiProtocol mAddressType;
         std::string mFullType;
-        uint8_t mAttempts = 0;
+        uint8_t mAttempts                = 0;
+        AvahiServiceResolver * mResolver = nullptr;
+
+        ~ResolveContext()
+        {
+            if (mResolver != nullptr)
```

Comment:
after calling avahi_service_resolver_free(mResolver); it is good to set mResolver as nullptr as well

Reply:
Updated, although still unsure because this is in the destructor, so anything accessing it would be accessing invalid memory right away. I guess this would more easily catch invalid access with segfaults.